pkg/kubectl/cmd/util/factory.go

```diff
@@ -355,9 +405,9 @@ func (f *factory) DiscoveryClient() (discovery.CachedDiscoveryInterface, error)
 	return NewCachedDiscoveryClient(discoveryClient, cacheDir, time.Duration(10*time.Minute)), nil
 }
 
-func (f *factory) Object() (meta.RESTMapper, runtime.ObjectTyper) {
+func Object(dcf DiscoveryClientFactory, ccf ClientCacheFactory) (meta.RESTMapper, runtime.ObjectTyper) {
```

Comment:
I'm against these changes.  We removed exactly this behavior this past release to make the code easier to follow and easier to break up later.  This feels like a step backwards.

@AdoHe @fabianofranz @kubernetes/sig-cli